docs/building/forms.md

```diff
@@ -21,36 +21,61 @@ Form
-    description - (string) the description of this FormItem
-    header      - (string) if provided will overwrite the Form.header
-    footer      - (string) if provided will overwrite the Form.footer
-
-FormItemMenu
-    type        - (string) defaults to "form-menu"
-    name        - (string) the name of this FormItemMenu, used in form serialization
-    header      - (string) if provided will overwrite the Form.header
-    footer      - (string) if provided will overwrite the Form.footer
-    body        - (array) FormItemMenuItem objects - see below
-
-
-FormItemMenuItem
-    type        - (string) defaults to "menu"
-    value       - (string) the value of this FormItemMenuItem, used in form
-                           serialization
+    type - (string) enum: "string", "hidden", "int", "float", "date", "datetime", "form-menu"
+        "string"    - the user should enter a string during this step
+        "hidden"    - will not be displayed to the user
+        "int"       - the user should enter a valid number
+        "float"     - the user could enter a floating number
+        "date"      - the user should enter a date
+        "datetime"  - the user should enter a date and time
+        "form-menu" - the user should choose an option from the menu
+
+    name                       - (string) the name of this FormItem, used in form serialization
+
+    body                       - (array of objects) MenuItemFormItem objects - see below - required only for type=form-menu
+    chunking_footer            - (string) shown in the footer of the sms chunks
+    confirmation_label         - (string) shown in the confirmation menu
+    description                - (string) the description of this FormItem
+    editable                   - (bool) indicates this form item is editable or not
+    footer                     - (string) if provided will overwrite the Form.footer
+    header                     - (string) if provided will overwrite the Form.header
+    max_length                 - (int) validates the user input - for type=string
+    max_length_error           - (string) message to be shown on max_length error
+    max_value                  - (int) validates the user input - for type=int|float
+    max_value_error            - (string) message to be shown on max_value error
+    meta                       - (object) MenuFormItemMeta object - see below - applies only for type=form-menu
+    method                     - (string) http method, how the callback url should be triggered
+    min_length                 - (int) validates the user input - for type=string
+    min_length_error           - (string) message to be shown on min_length error
+    min_value                  - (int) validates the user input - for type=int|float
+    min_value_error            - (string) message to be shown on min_value error
+    required                   - (bool) can be skipped if set to false
```

Comment:
This means we support SKIP command, or how does a not required step look like?

Reply:
Yes, already supported by the wizard engine ..